include/Context.hpp

```diff
@@ -0,0 +1,58 @@
+#ifndef CONTEXT_HPP
+#define CONTEXT_HPP
+
+#include <QAbstractListModel>
+#include <QString>
+#include <QUuid>
+#include <QObject>
+
+class Song : public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY(QString name MEMBER m_name READ getName);
+    Q_PROPERTY(QString author MEMBER m_author READ getAuthor);
+    Q_PROPERTY(QUuid id MEMBER m_id CONSTANT READ getUuid)
+
+public:
+    Song(QString name = QString(), QString author = QString(), QUuid id = QUuid(), QObject *parent = nullptr);
+    Song(const Song &s);
+    Song &operator=(const Song &other);
+    void setName(QString &name) { m_name = name; };
```

Comment:
Refactoring: Don't mess up class implementation with the class declaration. Handle it separately.